pkg/storegateway/bucket.go

```diff
@@ -202,6 +205,13 @@ func WithQueryGate(queryGate gate.Gate) BucketStoreOption {
 	}
 }
 
+// WithReaderGate sets a readerGate to use instead of a noopGate.
```

Comment:
```suggestion
// WithReaderGate sets a readerGate to use instead of gate.NewNoop().
```

this makes it a bit more clear which implementation you are referring to. Some IDEs also allow to jump to that definition form the comment.